docs/src/BrandingCssVarsProvider.tsx

```diff
@@ -53,31 +56,96 @@ const themeOptions = {
 };
 
-// `createColorScheme` is available in Material UI v6+
-// TODO: use the `createTheme` once the MUI X repo upgrade to Material UI v6+
-const theme =
-  typeof createColorScheme === 'function'
-    ? createTheme({
-        cssVariables: {
-          cssVarPrefix: 'muidocs',
-          colorSchemeSelector: 'data-mui-color-scheme',
-        },
-        ...themeOptions,
-      })
-    : extendTheme({
-        cssVarPrefix: 'muidocs',
-        colorSchemeSelector: 'data-mui-color-scheme',
-        ...themeOptions,
-      });
-
 // TODO: use the `ThemeProvider` once the MUI X repo upgrade to Material UI v6+
 const ThemeVarsProvider = typeof createColorScheme === 'function' ? ThemeProvider : CssVarsProvider;
 
-export default function BrandingCssVarsProvider(props: { children: React.ReactNode }) {
-  const { children } = props;
+export function setDocsColors(primary: Record<string, string>, secondary: Record<string, string>) {
+  function injectPalette(prefix: string, palette: string, color: string) {
+    // simplified logic of `createPalette` to avoid `useTheme`.
+    const light = lighten(color, 0.2);
+    const dark = darken(color, 0.3);
+    const contrastText = getContrastRatio(color, '#fff') >= 3 ? '#fff' : 'rgba(0, 0, 0, 0.87)';
+
+    document.documentElement.style.setProperty(`--${prefix}-palette-${palette}-main`, color);
+    document.documentElement.style.setProperty(
+      `--${prefix}-palette-${palette}-mainChannel`,
+      colorChannel(color),
+    );
+    document.documentElement.style.setProperty(`--${prefix}-palette-${palette}-light`, light);
+    document.documentElement.style.setProperty(
+      `--${prefix}-palette-${palette}-lightChannel`,
+      colorChannel(light),
+    );
+    document.documentElement.style.setProperty(`--${prefix}-palette-${palette}-dark`, dark);
+    document.documentElement.style.setProperty(
+      `--${prefix}-palette-${palette}-darkChannel`,
+      colorChannel(dark),
+    );
+    document.documentElement.style.setProperty(
+      `--${prefix}-palette-${palette}-contrastText`,
+      contrastText,
+    );
+    document.documentElement.style.setProperty(
+      `--${prefix}-palette-${palette}-contrastTextChannel`,
+      colorChannel(contrastText),
+    );
+  }
+  if (typeof document !== 'undefined') {
+    injectPalette('muidocs', 'primary', primary.main);
+    injectPalette('muidocs', 'secondary', secondary.main);
+
+    ['50', '100', '200', '300', '400', '500', '600', '700', '800', '900'].forEach((key) => {
+      document.documentElement.style.setProperty(`--muidocs-palette-primary-${key}`, primary[key]);
+      document.documentElement.style.setProperty(
+        `--muidocs-palette-secondary-${key}`,
+        secondary[key],
+      );
+    });
+
+    injectPalette('mui', 'primary', primary.main);
+    injectPalette('mui', 'secondary', secondary.main);
+  }
+}
+
+export function resetDocsColor() {
+  if (typeof document !== 'undefined') {
+    document.documentElement.removeAttribute('style');
```

Comment:
Are we sure there are no other styles added on this element? Would id be better to create a style tag with specific id and remove it instead?

Reply:
Yep, it's a quick approach to test if it works.